File: bionic-js-tool/testing-code/swift/project-with-host-files/HostProject/host/libs/FuelType.swift
import JavaScriptCore
import Bjs

class FuelType: BjsObject {
    
    class var Electricity:FuelType? {
        get {
            return bjs.getObj(bjs.getProperty(self.bjsClass, "Electricity"), FuelType.bjsFactory)
        }
    }
    
    class var NaturalGas:FuelType? {
        get {
            return bjs.getObj(bjs.getProperty(self.bjsClass, "NaturalGas"), FuelType.bjsFactory)
        }
    }
    
    class var Diesel:FuelType? {
        get {
            return bjs.getObj(bjs.getProperty(self.bjsClass, "Diesel"), FuelType.bjsFactory)
        }
    }
    
    class var Petrol:FuelType? {
        get {
            return bjs.getObj(bjs.getProperty(self.bjsClass, "Petrol"), FuelType.bjsFactory)
        }
    }
    
    class var Kerosene:FuelType? {
        get {
            return bjs.getObj(bjs.getProperty(self.bjsClass, "Kerosene"), FuelType.bjsFactory)
        }
    }
    
    var name:String? {
        get {
            return FuelType.bjs.getString(bjsGetProperty("name"))
        }
    }
    
    var cost:Double? {
        get {
            return FuelType.bjs.getFloat(bjsGetProperty("cost"))
        }
    }
    
    private static var _bjsLocator: BjsLocator = BjsLocator("BeautifulVehicles", "FuelType")
    override class var bjsLocator: BjsLocator { _bjsLocator }
    class func bjsFactory(_ jsObject: JSValue) -> FuelType { FuelType(jsObject) }
}

File: bionic-js-tool/testing-code/java/project-with-host-files/HostProject/src/motor-vehicles/java/test/project/host/libs/FuelType.java
package test.project.host.libs;

import jjbridge.api.runtime.JSReference;
import jjbridge.api.value.strategy.FunctionCallback;
import bionic.js.Bjs;
import bionic.js.BjsTypeInfo;
import bionic.js.BjsObjectTypeInfo;
import bionic.js.Lambda;
import java.util.Date;
import bionic.js.BjsObject;

@BjsTypeInfo.BjsLocation(project = "BeautifulVehicles", module = "FuelType")
public class FuelType extends BjsObject {
    
    protected <T extends BjsObject> FuelType(Class<T> type, JSReference jsObject) {
        super(type, jsObject);
    }
    
    protected <T extends BjsObject> FuelType(Class<T> type, JSReference[] arguments) {
        super(type, arguments);
    }
    
    public FuelType(JSReference jsObject) {
        this(FuelType.class, jsObject);
    }
    
    public static FuelType Electricity() {
        return bjs.getObj(bjs.getProperty(bjsClass, "Electricity"), FuelType.bjsFactory, FuelType.class);
    }
    
    public static FuelType NaturalGas() {
        return bjs.getObj(bjs.getProperty(bjsClass, "NaturalGas"), FuelType.bjsFactory, FuelType.class);
    }
    
    public static FuelType Diesel() {
        return bjs.getObj(bjs.getProperty(bjsClass, "Diesel"), FuelType.bjsFactory, FuelType.class);
    }
    
    public static FuelType Petrol() {
        return bjs.getObj(bjs.getProperty(bjsClass, "Petrol"), FuelType.bjsFactory, FuelType.class);
    }
    
    public static FuelType Kerosene() {
        return bjs.getObj(bjs.getProperty(bjsClass, "Kerosene"), FuelType.bjsFactory, FuelType.class);
    }
    
    public String name() {
        return bjs.getString(bjsGetProperty("name"));
    }
    
    public Double cost() {
        return bjs.getDouble(bjsGetProperty("cost"));
    }
    
    private static final JSReference bjsClass = BjsObjectTypeInfo.get(FuelType.class).bjsClass();
    public static final Bjs bjs = BjsObjectTypeInfo.get(FuelType.class).bjsLocator.get();
    public static final Bjs.JSReferenceConverter<FuelType> bjsFactory = FuelType::new;
}